precompiles/gov/tx.go

```diff
@@ -87,3 +94,90 @@ func (p Precompile) VoteWeighted(
 
 	return method.Outputs.Pack(true)
 }
+
+// Deposit defines a method to add a deposit to a proposal
+func (p Precompile) Deposit(
+	ctx sdk.Context,
+	origin common.Address,
+	contract *vm.Contract,
+	stateDB vm.StateDB,
+	method *abi.Method,
+	args []interface{},
+) ([]byte, error) {
+	msg, depositorAddr, err := NewMsgDeposit(method, args)
+	if err != nil {
+		return nil, err
+	}
+
+	// Deposit must have length 1
+	if len(msg.Amount) != 1 {
+		return nil, errors.New(ErrInvalidDeposit)
+	}
+
+	// If the contract is the depositor, we don't need an origin check
+	// Otherwise check if the origin matches the depositor address
+	isOriginDepositor := contract.CallerAddress == origin
+	isContractDepositor := contract.CallerAddress == depositorAddr && !isOriginDepositor
+	if !isContractDepositor && origin != depositorAddr {
+		return nil, fmt.Errorf(ErrDifferentOriginDepositor, origin.String(), depositorAddr.String())
+	}
+
+	msgSrv := govkeeper.NewMsgServerImpl(&p.govKeeper)
+	if _, err = msgSrv.Deposit(ctx, msg); err != nil {
+		return nil, err
+	}
+
+	if err = p.EmitDepositEvent(ctx, stateDB, msg.ProposalId, depositorAddr, msg.Amount); err != nil {
+		return nil, err
+	}
+
+	if !isOriginDepositor {
+		// NOTE: This ensures that the changes in the bank keeper are correctly mirrored to the EVM stateDB
+		// when calling the precompile from a smart contract
+		// This prevents the stateDB from overwriting the changed balance in the bank keeper when committing the EVM state.
+		// Need to scale the amount to 18 decimals for the EVM balance change entry
+
+		scaledAmt := evmtypes.ConvertAmountTo18DecimalsBigInt(msg.Amount[0].Amount.BigInt())
+		p.SetBalanceChangeEntries(cmn.NewBalanceChangeEntry(depositorAddr, scaledAmt, cmn.Sub))
```

Comment:
The deposit is transferring funds to a module, adding only the change entry for the sender should result in a mint of the same amount for the gov module:

```
	// update the governance module's account coins pool
	err = keeper.bankKeeper.SendCoinsFromAccountToModule(ctx, depositorAddr, types.ModuleName, depositAmount)
	if err != nil {
		return false, err
	}

```